skrub/_string_encoder.py

```diff
@@ -152,6 +157,8 @@ def fit_transform(self, X, y=None):
             # Therefore, self.n_components_ below stores the resulting
             # number of dimensions of result.
             result = X_out[:, : self.n_components].toarray()
+            result = result.copy()
```

Comment:
I suppose the `copy()` is that otherwise the reference to the slice would prevent `X_out` from being garbage collected; might be worth a short comment